Updates/0.3.3.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\QueuedTracking;
+
+use Piwik\Container\StaticContainer;
+use Piwik\Plugins\QueuedTracking\Queue\Manager;
+use Piwik\Updater;
+use Piwik\Updates;
+
+class Updates_0_3_3 extends Updates
+{
+    /**
+     * @var Manager
+     */
+    protected $queueManager;
+
+    public function doUpdate(Updater $updater)
+    {
+        $settings = $this->getSettings();
+        $numQueueWorkers = $settings->numQueueWorkers->getValue();
+
+        if ($numQueueWorkers <= 1) {
+            return; // it is pointless to rewrite one queue to the same queue
+        }
+
+        $this->rewriteQueues($numQueueWorkers);
```

Comment:
I resigned from using `moveRequestsIntoAnotherQueue()` because update script run into endless loop while rewriting queues. This is because `moveRequestsIntoAnotherQueue()` is designed for use case when number of queues is decreased, let's say from 4 to 2. In such case queues 3 and 4 are rewritten to 1 and 2 until there's nothing left. Clearing 3 and 4 is possible because no new requests go there even when new tracking requests appear from the website.

In case of changing distributing algorithm we rewrite to the same number of queues. In such case no queue is cleared and `moveRequestsIntoAnotherQueue()` fetches new portion of requests endlessly. So to make rewriting possible to finish I load and clear all queues at once and then redistribute requests.

Risk of getting into trouble while loading all queues may be decreased by loading one queue at the time, redistributing it, then loading next queue and so on. I implemented loading all queues at once because I thought otherwise there may be problems with order of requests, but now I don't see such problem.